docs/architecture.rst

```diff
@@ -1,142 +1,211 @@
-dig into provider-specific APIs.
-
-* **Folder**: Represents a specified location (could be local, remote, or file storage), for managing where various
-  Runhouse resources live.
-
-* **Blob**: Represents a data object stored in a particular system.
-
-* **File**: Represents a file object stored in a particular system.
+  .. code-block:: python
 
-Accessibility and Management
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    import runhouse as rh
 
-Runhouse enables the portability and sharing of resources across users and environments, and provides
-tools for visibility and management of these resources as long-living assets shared by teams or projects.
+    # [1] and [2]
+    gpu = rh.cluster(name="rh-a10x", instance_type="A10G:1", provider="aws").up_if_not()
 
-Accessibility
--------------
+    # [3]
+    sd_env = rh.env(reqs=["torch", "transformers", "diffusers"], name="sd_generate")
+    remote_sd_generate = rh.function(sd_generate).to(gpu, worker=sd_env)
 
-The Runhouse RNS (Resource Naming System) provides a convenient way to name, persist, and recall resources
-acoss environments. Meanwhile, the Secrets APIs provide a simple interface for storing and retrieving secrets
-from your secure Runhouse account.
+    # [4]
+    imgs = remote_sd_generate("A hot dog made out of matcha.")
+    imgs[0].show()
 
-* **Resource Naming System (RNS)**: Consists of lightweight metadata for each resource type to captures the
-  information needed to load it in a new environment, and a mechanism for saving and loading from either the working
-  git repo or a remote Runhouse key-value metadata store. The git-based approach (Local RNS) allows for local
-  persistence and versioning or sharing across OSS projects. The metadata store (Runhouse RNS) is even more portable;
-  it allows resource sharing across users and environments, anywhere there is an Internet connection and Python
-  interpreter. The RNS is backed by a management API (see below) to view and manage all resources.
+    # [5]
+    remote_sd_generate.save()
+    sd_upsampler = rh.function(name="/my_username/sd_upsampler")
+    high_res_imgs = sd_upsampler(imgs)
 
-* **Secrets API**: Provides a simple interface for storing and retrieving secrets to a allow a more seamless
-  experience when accessing resources across environments. It provides a simple interface for storing and retrieving
-  secrets from a variety of providers (e.g. AWS, Azure, GCP, Hugging Face, Github, etc.) as well as SSH Keys and
-  custom secrets, and stores them in Hashicorp Vault.
+    # [6]
+    gpu.down()
 
-* **Configs**: Set and preserve default configs across environments.
+  .. note::
 
-Management
-----------
 
-Runhouse provides tools for visibility and management of resources as long-living assets shared by teams or projects.
+1. **Specify and/or Allocate Compute**: Runhouse can allocate compute to the application on the fly, either by
```

Comment:
UX suggestion: Use H3's for these instead of bullets so they show up in the Table of Contents for the page. Will also make the sublist in "Starting the daemon" just a normal list.